src/backends/perf_pt/collect.c

```diff
@@ -265,7 +265,22 @@ open_perf(pid_t target_tid) {
     attr.precise_ip = 3;
 
     // Acquire file descriptor through which to talk to Intel PT.
-    ret = syscall(SYS_perf_event_open, &attr, target_tid, -1, -1, 0);
+    //
+    // This syscall could return EBUSY, meaning another process or thread has
+    // locked the Perf device. We spin around until we get an fd.
+    while (true) {
+        ret = syscall(SYS_perf_event_open, &attr, target_tid, -1, -1, 0);
+        if (ret == -1) {
```

Comment:
I think this entire `if` is equivalent to:

```c
if (ret == -1 && errno == EBUSY) {
    continue;
} else {
    break;
}
```

Reply:
Where do you catch the other `-1` cases that `perf_event_open` could return?

Reply:
I don't, but nor does the commit: if it gets `-1` and it's not `EBUSY` it `goto clean`s.

Reply:
In this PR or ...? Assuming you mean in a later PR, I'd suggest we simplify it as I suggested for the time being because then when you do come to change it, it'll be clearer what's going on. At the moment it is a bit confusing.